### PR TITLE
fix(ivy): ExpressionChangedAfterItHasBeenCheckedError for SafeValue

### LINKED ISSUE #33448: [Ivy] ExpressionChangedAfterItHasBeenCheckedError when using bypassSecurityTrustStyle in a property getter
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/core

### Is this a regression?

 Yes, it works in non-ivy mode

### Description

 A clear and concise description of the problem...

```
  <div [style.background-image]="iconSafe"></div>

   ...

  get iconSafe() {
    return this.sanitizer.bypassSecurityTrustStyle(`url(${this.icon})`);
  }
```

```
core.js:6047 ERROR Error: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked. Previous value: 'SafeValue must use [property]=binding: url(https://i.imgur.com/4AiXzf8.jpg) (see http://g.co/ng/security#xss)'. Current value: 'SafeValue must use [property]=binding: url(https://i.imgur.com/4AiXzf8.jpg) (see http://g.co/ng/security#xss)'.
```

## 🔬 Minimal Reproduction

 

https://github.com/fr0/angular-ivy-test2

1. `npm install`
1. `ng serve`
1. open http://localhost:4200 and press F12, observe error in console

## 🔥 Exception or Error
<pre><code>
core.js:6047 ERROR Error: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked. Previous value: 'SafeValue must use [property]=binding: url(https://i.imgur.com/4AiXzf8.jpg) (see http://g.co/ng/security#xss)'. Current value: 'SafeValue must use [property]=binding: url(https://i.imgur.com/4AiXzf8.jpg) (see http://g.co/ng/security#xss)'.
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 9.0.0-next.16
Node: 12.12.0
OS: darwin x64
Angular: 9.0.0-next.14
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-next.16
@angular-devkit/build-angular     0.900.0-next.16
@angular-devkit/build-optimizer   0.900.0-next.16
@angular-devkit/build-webpack     0.900.0-next.16
@angular-devkit/core              9.0.0-next.16
@angular-devkit/schematics        9.0.0-next.16
@angular/cli                      9.0.0-next.16
@ngtools/webpack                  9.0.0-next.16
@schematics/angular               9.0.0-next.16
@schematics/update                0.900.0-next.16
rxjs                              6.5.3
typescript                        3.6.4
webpack                           4.41.2

</code></pre>

### PR DESCRIPTION
**Pull request by mhevery**
Fix #33448

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: packages/core/src/render3/util/styling_utils.ts [hunk 1]
```diff
@@ -170,6 +171,14 @@ export function getPropValuesStartPosition(
   return startPosition;
 }
 
+export function hasValueChangedUnwrapSafeValue(
+    a: NO_CHANGE | StylingMapArray | number | String | string | null | boolean | undefined | {},
+    b: NO_CHANGE | StylingMapArray | number | String | string | null | boolean | undefined |
```
Comment (alfaproject) [inline packages/core/src/render3/util/styling_utils.ts]: `String` and `string` are the same. I suggest to pick `string`

Comment (mhevery) [reply packages/core/src/render3/util/styling_utils.ts]: They are not same: http://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=1&pc=39#code/G4QwTgBKA2CuCmAuCBnALmAlgOwOYQF4Jt4B3CAZQx1wAoByEAIwGN6BKAbiA

But, you are right `String` is there for historical reasons, and we should get rid of it. Will do it in separate PR.

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:279980249:BASE:279980290:1573577852463:9fdefeeb)
[ivy presubmit](http://test/OCL:279980249:BASE:279980577:1573577958840:e4e4b057)

File: packages/core/src/render3/util/styling_utils.ts (hunk 1 above)
Comment (mhevery) [reply packages/core/src/render3/util/styling_utils.ts]: [style: Remove used of `String` as type and use `string` instead. by mhevery · Pull Request #33763 · angular/angular](https://github.com/angular/angular/pull/33763)

File: packages/core/test/acceptance/styling_spec.ts [hunk 2]
```diff
@@ -2337,6 +2337,25 @@ describe('styling', () => {
 
         function splitSortJoin(s: string) { return s.split(/\s+/).sort().join(' ').trim(); }
       });
+
+  describe('ExpressionChangedAfterItHasBeenCheckedError', () => {
+    it('should not throw when bound to bypass security value', () => {
```
Comment (IgorMinar) [inline packages/core/test/acceptance/styling_spec.ts]:
```suggestion
    it('should not throw when bound to SafeValue', () => {
```

**Review by IgorMinar — APPROVED** [review]
lgtm, but you might want to handle innerHtml in the same way - currently we ignore/don't check SafeValues there.

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM after Igor comments are addressed

File: packages/core/src/render3/util/styling_utils.ts (hunk 1 above)
Comment (pkozlowski-opensource) [reply packages/core/src/render3/util/styling_utils.ts]: @IgorMinar this was already addressed in https://github.com/angular/angular/pull/33763 - this PR just need a rebase.

**Pull request closed without merging**